docs/dev/framework/search-indexing.md

```diff
@@ -148,3 +150,59 @@ $jsonLdScriptsData =  $document->extractJsonLdScripts('https://schema.org', 'Pro
 The variable `$jsonLdScriptsData` now contains an already decoded array of all JSON LD scripts with `@context` equal to
 `https://schema.org` and `@type` equal to `Product`. You can conveniently extract the product price and index it along
 with all the other data you need.
+
+
+## Dynamically Disabling Indexing for Pages
+
+Within the back end, you can disable search indexing for specific pages in the settings of each page. In some cases, you
+might want to dynamically disable indexing under certain conditions for arbitrary pages. To disable indexing, you need
+to set the `noSearch` property of the `PageModel` object of the current page to `true`, before the page's content is added 
+to the index.
+
+There are multiple ways to achieve this. One way is to use the [generatePage][generatePage] hook:
+
+```php
+// src/EventListener/GeneratePageListener.php
+namespace App\EventListener;
+
+use Contao\CoreBundle\ServiceAnnotation\Hook;
+use Terminal42\ServiceAnnotationBundle\ServiceAnnotationInterface;
+
+/**
+ * @Hook("generatePage")
+ */
+class GeneratePageListener implements ServiceAnnotationInterface
+{
+    public function __invoke(\Contao\PageModel $pageModel): void
+    {
+        if (/* … */) {
+            $pageModel->noSearch = true;
+        }
+    }
+}
+```
+
+Starting with Contao **4.6** you could also fetch the `pageModel` from the request attributes (if present) in your own 
+`kernel` event listener.
+
+
+## Ignoring GET Parameters
```

Comment:
I wouldn't document that because it's actually just plain wrong that this even works.
It will get indexed which is correct. If you don't want this to happen, just set the page to `noSearch` or `noindex` depending on the GET parameters set.

Reply:
Well, if you have a page `https://example.org/foo.html` and a bot calls `https://example.org/foo.html?foobar=foo`, then you will get two search results. Not sure what the best way to deal with that is. I was merely documenting what already works.

Reply:
Yeah, and what I'm saying is that just because it works, doesn't mean it should.
We cannot remove that because of BC but you should certainly not use it.

You will always get multiple cache entries. It's easy to flood caches by just adding nonsense query parameters to URLs but that's just how caches work.

If you want to send a 404 specifically when `foobar=foo` does not exist in your module, you can.

Reply:
It's now removed :)